packages/core/helper-plugin/src/hooks/useRBAC.js

```diff
@@ -40,7 +40,12 @@ export const useRBAC = (permissionsToCheck, passedPermissions) => {
 
   const queryResults = useQueries(
     permissionsToCheckEntries.map(([name, permissions]) => ({
-      queryKey: ['useRBAC', name, permissions.map(({ action, subject }) => ({ action, subject }))],
+      queryKey: [
+        'useRBAC',
+        name,
+        permissions.map(({ action, subject }) => ({ action, subject })),
+        userPermissions.map(({ action, subject }) => ({ action, subject })),
```

Comment:
Why do we map here instead of passing in the full arrays?

Reply:
I haven't really thought about it, you're probably right though, if we just pass the variables then we're covered for everything... I suppose that's what you're eluding too?

Reply:
Yes, exactly :)